rs/boundary_node/rate_limits/canister_client/src/main.rs

```diff
@@ -36,17 +40,30 @@ async fn main() {
     // parse command-line arguments
     let cli = Cli::parse();
 
-    // create the agent
-    let identity = Secp256k1Identity::from_private_key(
-        SecretKey::from_sec1_pem(&cli.identity_key).expect("failed to parse the identity key"),
-    );
-    let agent = Agent::builder()
-        .with_url(cli.ic_domain)
-        .with_identity(identity)
-        .build()
-        .expect("failed to build the agent");
-
-    submit_config(cli.config_file, cli.canister_id, agent, cli.debug)
-        .await
-        .expect("failed to submit the config to the canister");
+    if cli.check {
+        match check_config(cli.config_file) {
```

Comment:
I think you can make the `tokio::main` return a `Result` and use normal error handling here, with `context` and whatnot.

Reply:
I tried to do that